### PR TITLE
Document gcloud authentication

### LINKED ISSUE #1938: Document gcloud for 2.0
This issue is about <https://circleci.com/docs/1.0/google-auth/> (source file: <https://github.com/circleci/circleci-docs/blob/master/jekyll/_cci1/google-auth.md>)

We started recommending that folks use [gcloud firebase](https://support.circleci.com/hc/en-us/articles/360000028928) for using the android emulator. The most complex part of this might be the auth. It would be worth adding a similar section from our 1.0 docs to the 2.0 docs.

### PR DESCRIPTION
**Pull request by smart-alek**
This is a rewrite of the 1.0 article on [authenticating Google Cloud Platform](https://circleci.com/docs/1.0/google-auth/). This is particularly relevant for Android users, for whom we've been recommending [gcloud firebase](https://support.circleci.com/hc/en-us/articles/360000028928).

I've linked this article in two locations: the Android language guide, and Google Cloud section of the Deploys page. I think it should also live on an index page somewhere, but I'm not sure it fits in any existing categories.

Fixes #1938.

### CODE REVIEW

File: jekyll/_cci2/deployment_integrations.md [hunk 1]
```diff
@@ -177,8 +177,11 @@ Refer to the full example in the [2.0 Project Tutorial]( {{ site.baseurl }}/2.0/
 
 ## Google Cloud
 
-Ensure that the Google Cloud SDK is installed in your primary container so that `gcloud` and all of the necessary tools for manipulating Kubernetes resources are at your disposal inside your deployment script/commands. In the following example, if the build passes and the current branch was the master branch, CircleCI 
-runs `deploy.sh` to do the actual deployment work.
+Before deploying to Google Cloud, ensure that you have authenticated the `gcloud` command line tool. To do that, you can read the [Authenticating Google Cloud Platform]({{ site.baseurl }}/2.0/) document.
```
Comment (michelle-luna) [inline jekyll/_cci2/deployment_integrations.md]: this should link to  /2.0/google-auth.md

Comment (michelle-luna) [reply jekyll/_cci2/deployment_integrations.md]: I think this section should also link to /2.0/google-container-engine.md

File: jekyll/_cci2/google-auth.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Before you can use the `gcloud` command line tool with CircleCI, you must authenticate it. To do this, you will need to create a [service account][]. You can then add this account as an [environment variable]({{ site.baseurl }}/2.0/env-vars/) within CircleCI. Your build script will decode this variable and authenticate the `gcloud` tool for use in your project.
+
+## Prerequisites
+
+- A Google account.
+- A Google Cloud Platform project.
+
+## Steps
+
```
Comment (michelle-luna) [inline jekyll/_cci2/google-auth.md]: Add a summary sentence of the result of completing these steps after the heading.

File: jekyll/_cci2/google-auth.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+### Create a service account and download
```
Comment (michelle-luna) [inline jekyll/_cci2/google-auth.md]: Use initial caps in headings for words with more than three letters (generally)

File: jekyll/_cci2/google-auth.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+Go to Google's [Getting Started with Authentication][] article and follow the instructions in the **Creating a service account** section.
+
+### Add service account to CircleCI environment
```
Comment (michelle-luna) [inline jekyll/_cci2/google-auth.md]: initial caps

File: jekyll/_cci2/google-auth.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+1. Encode the JSON file you downloaded in base64 format and copy the result to the clipboard.
+2. In the CircleCI application, go to your project's settings by clicking the gear icon in the top right.
+3. In the **Build Settings** section, click **Environment Variables**, then click the **Add Variable** button.
+4. Name the variable. In this example, the variable is named `$GCLOUD_SERVICE_KEY`.
+5. Paste the contents from step 2 into the **Value** field.
```
Comment (michelle-luna) [inline jekyll/_cci2/google-auth.md]: step 2 -> Step 2

File: jekyll/_cci2/language-android.md [hunk 6]
```diff
@@ -18,7 +18,7 @@ There are some assumptions made in this guide:
...
+Running the Android emulator is not currently supported on CircleCI 2.0. To run emulator tests from a job, consider using an external service like [Firebase Test Lab](https://firebase.google.com/docs/test-lab). To do this, you can use the [gcloud command line](https://firebase.google.com/docs/test-lab/command-line). Google Cloud tools are preinstalled in our Android Docker images, which you can find on [GitHub](https://github.com/CircleCI-Public/circleci-dockerfiles/tree/master/android/images) or [Docker Hub](https://hub.docker.com/r/circleci/android/tags). Before using `gcloud`, ensure that you have authenticated it. See the [Authenticating Google Cloud Platform]({{ site.baseurl }}/2.0/) document for more details.
```
Comment (michelle-luna) [inline jekyll/_cci2/language-android.md]: change link to /2.0/google-auth.md

File: jekyll/_cci2/google-auth.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+6. Click the **Add Variable** button.
+
+### Decode and store service account
```
Comment (michelle-luna) [inline jekyll/_cci2/google-auth.md]: initial caps

File: jekyll/_cci2/google-auth.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+To authenticate the `gcloud` tool, you will first need to decode the environment variable you created above. You could do that by adding the following command to `config.yml`:
+
+    echo $GCLOUD_SERVICE_KEY | base64 --decode --ignore-garbage > ${HOME}/gcloud-service-key.json
+
+This will decode the secret into a file named `gcloud-service-key.json`.
+
+### Authenticate the `gcloud` tool
```
Comment (michelle-luna) [inline jekyll/_cci2/google-auth.md]: initial caps

File: jekyll/_cci2/google-auth.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+Update `gcloud`, authenticate, and set the project's active configuration.
+
+    sudo /opt/google-cloud-sdk/bin/gcloud --quiet components update
+    sudo /opt/google-cloud-sdk/bin/gcloud auth activate-service-account --key-file=${HOME}/gcloud-service-key.json
+    sudo /opt/google-cloud-sdk/bin/gcloud config set project $GCLOUD_PROJECT
+
+### Set Google application credentials
```
Comment (michelle-luna) [inline jekyll/_cci2/google-auth.md]: initial caps

File: jekyll/_cci2/google-auth.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
+---
+layout: classic-docs
+title: Authenticating Google Cloud Platform
+description: Authenticating Google Cloud Platform
+---
+
```
Comment (michelle-luna) [inline jekyll/_cci2/google-auth.md]: add: [Deploy]({{ site.baseurl }}/2.0/deployment_integrations/) > Authenticating Google Cloud Platform 

this is the navigation breadcrumb we should add for all new articles (changing the landing page link to wherever the top-level page lives

File: jekyll/_cci2/deployment_integrations.md (hunk 1 above)
Comment (smart-alek) [reply jekyll/_cci2/deployment_integrations.md]: whoops good catch!

File: jekyll/_cci2/google-auth.md (hunk 2 above)
Comment (smart-alek) [reply jekyll/_cci2/google-auth.md]: Isn't that what the overview is for?

**Pull request merged**